src/Cake.Core.Tests/Unit/IO/DirectoryPathTests.cs

```diff
@@ -255,6 +255,93 @@ public void Can_Not_Combine_Directory_Path_With_Absolute_Windows_Directory_Path(
             }
         }
 
+        public sealed class TheGetParentMethod
+        {
+            public sealed class InWindowsFormat
+            {
+                [WindowsTheory]
+                [InlineData("C:/Data", "C:/")]
+                [InlineData("C:/Data/Work", "C:/Data")]
+                [InlineData("C:/Data/Work/file.txt", "C:/Data/Work")]
+                [InlineData("../../Work", "../..")]
+                [InlineData("Data/../Work", "Data/..")]
+                [InlineData(@"\\A\B\C", @"\\A\B")]
+                [InlineData(@"\\A\B\c.txt", @"\\A\B")]
+                public void Should_Return_Parent_Directory(string directoryPath, string parentPath)
+                {
+                    // Given
+                    var path = new DirectoryPath(directoryPath);
+
+                    // When
+                    var result = path.GetParent();
+
+                    // Then
+                    Assert.Equal(parentPath, result.FullPath);
+                }
+
+                [WindowsTheory]
+                [InlineData("C:/")]
+                [InlineData("C:")]
+                [InlineData(@"\\A\")]
+                [InlineData(@"\\A")]
+                [InlineData("..")]
+                [InlineData("/..")]
+                [InlineData("/../")]
+                public void Should_Return_Null_If_No_Parent(string directoryPath)
```

Comment:
I feel there are cases for UNC paths missing:
* the parent of `\\server\share\folder` should be `\\server\share` <-- while this is obvious and probably already covered, I'd put it in, nonetheless.
* the parent of `\\server\share` is definitively missing: While the parent currently probably is `\\server` I'm not entirely sure that this is "correct" - `\\server` is not a directory that can easily accessed. I *think* the parent of `\\server\share` should be null. 

Any opinions on the last part (i.e. parent of `\\server\share`)? /cc @augustoproiete @patriksvensson